spherized_profiles/scripts/nbconverted/spherize-batch-effects.py

```diff
@@ -80,15 +80,30 @@
         )
 
         print(profile_df.shape)
-        profile_df.head()
 
-        spherize_df = normalize(
-            profiles=profile_df,
-            features="infer",
-            meta_features="infer",
-            samples="Metadata_broad_sample == 'DMSO'",
-            method="whiten",
-        )
+        # The second batch has multiple cell lines
+        if batch == "2017_12_05_Batch2":
+            spherize_df = (
+                profile_df
+                .groupby("Metadata_cell_line")
```

Comment:
No need to edit but pretty sure you don't need this logic (grouping by cell line is trivially valid in batch 1 as well)

Reply:
Regarding spherizing overall, we will make a lot of changes over time which are likely to improve the quality:
1. Also group by timepoint and not just cell line because if we don't we might be effectively factoring out subspaces we care about
2. Figure out epsilon
3. Drop outliers prior to spherizing

Reply:
> Also group by timepoint and not just cell line because if we don't we might be effectively factoring out subspaces we care about

nice catch, i'll update this

> Figure out epsilon

I think i'll skip this one for data freeze version 1. It seems to far off in the future to wait on.

> Drop outliers prior to spherizing

Yep, we already do this. We're discussing in https://github.com/broadinstitute/lincs-cell-painting/issues/65#issuecomment-824322518 and once we decide there, I'll run this notebook again with all the suggested changes (I'll also add drop outliers to consensus signature generation notebook)

Reply:
Great! I agree you should skip `epslion` optimization, and it's great you're updating 1. as well

Reply:
And great that the outlier features will be gone! For dropping outlier samples, we need new functionality in `pycytominer`, right? See https://github.com/cytomining/pycytominer/issues/140

Reply:
pycytominer currently has a very crude outlier removal strategy ([here](https://github.com/cytomining/pycytominer/blob/a04397d9cd7e25828d2f24f986a3386a79e6193d/pycytominer/cyto_utils/features.py#L110)), which can be specified as an operation in `feature_select()`.

It should easily handle Michael's features as defined in #65, but you're right, the method needs to be improved in the future. Thanks for opening that pycytominer issue!

Reply:
IIUC, it needs to be a different operation altogether (a row filter, not a column filter; `drop_outlier_features` is the latter), but we can discuss this in https://github.com/cytomining/pycytominer/issues/140

All set here from my end.